src/ParallelDataCollector.cpp

```diff
@@ -475,14 +475,25 @@ namespace splash
         DCParallelGroup group;
         if (dataName)
         {
+            /** create `obj_name` as a group if not exists
+             *
+             * if the `obj_name` with the given `group_path` not exists than
```

Comment:
oh dang! how do we even know it is a group?
This could be very unexpected behaviour if a user actually just mismatched their `write` and `writeAttribute` call...